### PR TITLE
fix: Stop using PYTHONPATH to set up sys.path

### PR DESCRIPTION
**Pull request by philsc**
This patch is an experiment for setting up `sys.path` inside
the final Python process instead of letting the runtime do it
via PYTHONPATH.

The motivation here is to address these 2 issues:
* #382 
* #792 

Try it with:
```
$ bazel run //tools:wheelmaker -- --help
```
Look at the generated files.

### CODE REVIEW

File: python/private/common/entrypoint.tmpl.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+RUNFILES = runfiles.Create()
+RUNFILES_DIR = Path(RUNFILES.EnvVars()["RUNFILES_DIR"])
+MAIN = RUNFILES_DIR / "%MAIN_REPO%/%MAIN_SHORT_PATH%"
+
+IMPORTS = %IMPORTS%
+
+# Work around sys.path[0] escaping the sandbox by deleting it.
+# See https://github.com/bazelbuild/rules_python/issues/382 for more info.
+# TODO(phil): How do we distinguish between safe-path Python and non-safe-path
```
Comment (aignas) [inline python/private/common/entrypoint.tmpl.py:15]: For inspiration, see: https://github.com/bazelbuild/rules_python/blob/627830e0756e6f7c09f2a95561963121dff9c166/python/private/py_console_script_gen.py#L50

**Comment by aignas** [comment]
Do you think it would help #1174 as well?

File: python/private/common/entrypoint.tmpl.py (hunk 1 above)
Comment (philsc) [reply python/private/common/entrypoint.tmpl.py]: Super cool. Thank you!

**Comment by philsc** [comment]
> Do you think it would help #1174 as well?

Not as-is, no. But I think it _could_. In the `python/private/common/entrypoint.tmpl.py` file we could add an import hook that tries to resolve imports with the repository name.

**Comment by kwlzn** [comment]
hmm, I noticed this was closed without merging. any reason why? we are hitting this in our repo and would love to contribute to a proper fix. @philsc @rickeylev

**Comment by aignas** [comment]
The solution using the script bootstrap and making it the default is in #2521. See our official docs config settings page on how to enable it in the meantime.

**Pull request closed without merging**